include/uacpi/acpi.h

```diff
@@ -1003,6 +1003,131 @@ UACPI_PACKED(struct acpi_ecdt {
+};
+
+// Constants for acpi_dbg2_device_info->port_subtype
+// if acpi_dbg2_device_info->port_type == ACPI_DBG2_TYPE_1394.
+enum acpi_dbg2_1394_subtype {
+    ACPI_DBG2_SUBTYPE_1394_STANDARD = 0x0,
+};
+
+// Constants for acpi_dbg2_device_info->port_subtype
+// if acpi_dbg2_device_info->port_type == ACPI_DBG2_TYPE_USB.
+enum acpi_dbg2_usb_subtype {
+    ACPI_DBG2_SUBTYPE_USB_XHCI_DEBUG = 0x0,
+    ACPI_DBG2_SUBTYPE_USB_EHCI_DEBUG = 0x1,
+};
+
+UACPI_PACKED(struct acpi_dbg2 {
+    struct acpi_sdt_hdr hdr;
+    uacpi_u32 offset_dbg_device_info;
+    uacpi_u32 number_dbg_device_info;
+})
+UACPI_EXPECT_SIZEOF(struct acpi_dbg2, 44);
+
+UACPI_PACKED(struct acpi_dbg2_dbg_device_info {
+    uacpi_u8 revision;
+    uacpi_u16 length;
+    uacpi_u8 number_generic_address_registers;
+    uacpi_u16 namespace_string_length;
+    uacpi_u16 namespace_string_offset;
+    uacpi_u16 oem_data_length;
+    uacpi_u16 oem_data_offset;
+    uacpi_u16 port_type;
+    uacpi_u16 port_subtype;
+    uacpi_u16 rsvd;
+    uacpi_u16 base_address_register_offset;
+    uacpi_u16 address_size_offset;
+    // Note that the are several extra fields at variable offsets
+    // (i.e., base address register, address size, namespace string and OEM data).
+    // However, they do not necessarily follow immediately after this struct;
+    // hence, we do not add a flexible array member here.
+})
+UACPI_EXPECT_SIZEOF(struct acpi_dbg2_dbg_device_info, 22);
+
+// Bit flags for acpi_scpr->interrupt_type.
+#define ACPI_SPCR_INTERRUPT_TYPE_8259 0x1
+#define ACPI_SPCR_INTERRUPT_TYPE_IOAPIC 0x2
+#define ACPI_SPCR_INTERRUPT_TYPE_IOSAPIC 0x4
+#define ACPI_SPCR_INTERRUPT_TYPE_GIC 0x8
+#define ACPI_SPCR_INTERRUPT_TYPE_PLIC_APLIC 0x10
+
+// Bit flags for acpi_spcr->pci_flags.
+#define ACPI_SPCR_PCI_FLAGS_DO_NOT_DISABLE 0x1
+
+// Constants for terminal_type.
+enum acpi_spcr_terminal_type {
+    ACPI_SPCR_TERMINAL_TYPE_VT100 = 0,
+    ACPI_SPCR_TERMINAL_TYPE_EXTENDED_VT100 = 1,
+    ACPI_SPCR_TERMINAL_TYPE_VT_UTF8 = 2,
+    ACPI_SPCR_TERMINAL_TYPE_ANSI = 3,
+};
+
+UACPI_PACKED(struct acpi_spcr {
+    struct acpi_sdt_hdr hdr;
+    uacpi_u8 interface_type;
+    uacpi_u8 rsvd[3];
+    struct acpi_gas base_address;
+    uacpi_u8 interrupt_type;
+    uacpi_u8 irq;
+    uacpi_u32 gsi;
+    uacpi_u8 configured_baud_rate;
+    uacpi_u8 parity;
+    uacpi_u8 stop_bits;
+    uacpi_u8 flow_control;
+    uacpi_u8 terminal_type;
+    uacpi_u8 language;
+    uacpi_u16 pci_device_id;
+    uacpi_u16 pci_vendor_id;
+    uacpi_u8 pci_bus_number;
+    uacpi_u8 pci_device_number;
+    uacpi_u8 pci_function_number;
+    uacpi_u32 pci_flags;
+    uacpi_u8 pci_segment;
+    uacpi_u32 uart_clock_frequency;
+    uacpi_u32 precise_baud_rate;
+    uacpi_u16 namespace_string_length;
+    uacpi_u16 namespace_string_offset;
+    // Like DBG2, SPCR also has extra fields at variable offsets,
+    // namely the namespace string.
+    // Likewise, we do not add a flexible array member here.
```

Comment:
I feel like the comment here is a bit verbose, maybe just say `// char namespace_string[] at a variable offset below`